[CI] ExecutionVarsIntegrationTests testVars failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/orn4244tbrehc/tests/:x-pack:plugin:watcher:internalClusterTest/org.elasticsearch.xpack.watcher.test.integration.ExecutionVarsIntegrationTests/testVars

**Reproduction line:**
```
./gradlew ':x-pack:plugin:watcher:internalClusterTest' --tests "org.elasticsearch.xpack.watcher.test.integration.ExecutionVarsIntegrationTests.testVars" -Dtests.seed=5AEC95C870BDC9E8 -Dtests.locale=nl-BE -Dtests.timezone=America/Santa_Isabel -Druntime.java=20
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.watcher.test.integration.ExecutionVarsIntegrationTests&tests.test=testVars

**Failure excerpt:**
```
org.elasticsearch.action.search.SearchPhaseExecutionException: all shards failed

  at __randomizedtesting.SeedInfo.seed([5AEC95C870BDC9E8:8EA78D927D084DEB]:0)
  at org.elasticsearch.action.search.AbstractSearchAsyncAction.onPhaseFailure(AbstractSearchAsyncAction.java:729)
  at org.elasticsearch.action.search.AbstractSearchAsyncAction.executeNextPhase(AbstractSearchAsyncAction.java:419)
  at org.elasticsearch.action.search.AbstractSearchAsyncAction.onPhaseDone(AbstractSearchAsyncAction.java:761)
  at org.elasticsearch.action.search.AbstractSearchAsyncAction.onShardFailure(AbstractSearchAsyncAction.java:513)
  at org.elasticsearch.action.search.AbstractSearchAsyncAction$1.onFailure(AbstractSearchAsyncAction.java:350)
  at org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:59)
  at org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:71)
  at org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:27)
  at org.elasticsearch.action.ActionListenerResponseHandler.handleException(ActionListenerResponseHandler.java:48)
  at org.elasticsearch.action.search.SearchTransportService$ConnectionCountingHandler.handleException(SearchTransportService.java:620)
  at org.elasticsearch.transport.TransportService$UnregisterChildTransportResponseHandler.handleException(TransportService.java:1684)
  at org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleException(TransportService.java:1401)
  at org.elasticsearch.transport.TransportService$DirectResponseChannel.processException(TransportService.java:1537)
  at org.elasticsearch.transport.TransportService$DirectResponseChannel.sendResponse(TransportService.java:1512)
  at org.elasticsearch.transport.TaskTransportChannel.sendResponse(TaskTransportChannel.java:51)
  at org.elasticsearch.action.support.ChannelActionListener.onFailure(ChannelActionListener.java:37)
  at org.elasticsearch.action.ActionRunnable.onFailure(ActionRunnable.java:92)
  at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:28)
  at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)
  at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:983)
  at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
  at java.lang.Thread.run(Thread.java:1623)

  Caused by: org.elasticsearch.action.NoShardAvailableActionException: [node_s1][127.0.0.1:30012][indices:data/read/search[phase/query]]


```

--- a/x-pack/plugin/watcher/src/internalClusterTest/java/org/elasticsearch/xpack/watcher/test/integration/ExecutionVarsIntegrationTests.java
+++ b/x-pack/plugin/watcher/src/internalClusterTest/java/org/elasticsearch/xpack/watcher/test/integration/ExecutionVarsIntegrationTests.java
@@ -11,6 +11,7 @@ import org.elasticsearch.common.xcontent.support.XContentMapValues;
 import org.elasticsearch.plugins.Plugin;
 import org.elasticsearch.protocol.xpack.watcher.PutWatchResponse;
 import org.elasticsearch.xcontent.ObjectPath;
+import org.elasticsearch.xpack.core.watcher.history.HistoryStoreField;
 import org.elasticsearch.xpack.core.watcher.support.xcontent.XContentSource;
 import org.elasticsearch.xpack.core.watcher.transport.actions.execute.ExecuteWatchRequestBuilder;
 import org.elasticsearch.xpack.core.watcher.transport.actions.execute.ExecuteWatchResponse;
@@ -110,7 +111,6 @@ public class ExecutionVarsIntegrationTests extends AbstractWatcherIntegrationTes
         }
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/95077")
     public void testVars() throws Exception {
         PutWatchResponse putWatchResponse = new PutWatchRequestBuilder(client()).setId(watchId)
             .setSource(
@@ -136,6 +136,7 @@ public class ExecutionVarsIntegrationTests extends AbstractWatcherIntegrationTes
         flush();
         assertBusy(() -> {
             refresh();
+            ensureGreen(HistoryStoreField.DATA_STREAM);
             SearchResponse searchResponse = searchWatchRecords(builder -> {
                 // defaults to match all;
             });

